mdsuite/calculators/potential_of_mean_force.py

```diff
@@ -126,157 +132,253 @@ def __init__(self, **kwargs):
     @call
     def __call__(
         self,
+        rdf_data: Computation,
         plot=True,
-        data_range=1,
         savgol_order: int = 2,
         savgol_window_length: int = 17,
+        number_of_shells: int = 1,
     ):
         """
         Python constructor for the class
 
         Parameters
         ----------
+        rdf_data : Computation
+                RDF data to use in the computation.
         plot : bool (default=True)
                             Decision to plot the analysis.
-        data_range : int (default=500)
-                            Range over which the property should be evaluated.
-                            This is not applicable to the current analysis as
-                            the full rdf will be calculated.
         savgol_order : int
                 Order of the savgol polynomial filter
         savgol_window_length : int
                 Window length of the savgol filter.
+        number_of_shells : int
+                Number of shells to integrate through.
         """
 
+        if isinstance(rdf_data, Computation):
+            self.rdf_data = rdf_data
+        else:
+            self.rdf_data = self.experiment.run.RadialDistributionFunction(plot=False)
+
         self.plot = plot
         self.data_files = []
 
+        # set args that will affect the computation result
         self.args = Args(
-            savgol_order=savgol_order, savgol_window_length=savgol_window_length
+            savgol_order=savgol_order,
+            savgol_window_length=savgol_window_length,
+            number_of_bins=self.rdf_data.computation_parameter["number_of_bins"],
+            cutoff=self.rdf_data.computation_parameter["cutoff"],
+            number_of_configurations=self.rdf_data.computation_parameter[
+                "number_of_configurations"
+            ],
+            number_of_shells=number_of_shells,
         )
 
-    def _calculate_potential_of_mean_force(self):
+        # Auto-populate the results.
+        for i in range(self.args.number_of_shells):
+            self.result_keys.append(f"POMF_{i + 1}")
+            self.result_keys.append(f"POMF_{i + 1}_error")
```

Comment:
could also benefit from a common function

Reply:
I think when the calculators are restructured the way that recorded data is described in a calculator will be reassessed completely. For now I would leave this and change it when we change all of them.